src/components/Header.tsx

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import cn from 'classnames';
+import React, {
+  useRef, useEffect, useState, useContext, memo,
+} from 'react';
+import { addTodo } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { AuthContext } from './Auth/AuthContext';
+
+type Props = {
+  todos: Todo[],
+  isAllTodosCompleted: boolean,
+  setTemporaryTodo: (todo: Todo | null) => void,
+  showError: (message: string) => void,
+  setTodos: (prev: Todo[]) => void,
+  toggleAll: () => void,
+};
+
+export const Header: React.FC<Props> = memo(({
+  todos,
+  isAllTodosCompleted,
+  setTemporaryTodo,
+  showError,
+  setTodos,
+  toggleAll,
+}) => {
+  const [isAdding, setIsAdding] = useState(false);
+  const [title, setTitle] = useState('');
+
+  const user = useContext(AuthContext);
+
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    // focus the element with `ref={newTodoField}`
+    if (newTodoField.current) {
+      newTodoField.current.focus();
+    }
+  }, [isAdding]);
+
+  const onSubmitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (title) {
+      const todoForBack: Omit<Todo, 'id'> = {
+        userId: user?.id || 0,
+        title,
+        completed: false,
+      };
+
+      setTemporaryTodo({
+        id: 0,
+        ...todoForBack,
+      });
+
+      setIsAdding(true);
+
+      addTodo(todoForBack)
+        .then((newTodo) => {
+          setTodos([...todos, newTodo]);
```

Comment:
Never use `no-functional` setState for updating arrays.
```suggestion
          setTodos((prevTodos) => [...prevTodos, newTodo]);
```